pychemy/peptide_sets.py

```diff
@@ -106,3 +106,178 @@ def balanced_sets(proteins, max_set_size = 10, unique = 1):
+hopp_woods_hydrophobicity = {'A': -0.500,'C': -1.000,'D':  3.000,'E':  3.000,'F': -2.500,'G':  0.000,'H': -0.500,'I': -1.800,'K':  3.000,'L': -1.800,'M': -1.300,'N':  0.200,'P':  0.000,'Q':  0.200,'R':  3.000,'S':  0.300,'T': -0.400,'V': -1.500,'W': -3.400,'Y': -2.300 }
+kyte_doolittle_hydrophobicity = {'A':  1.800,'C':  2.500,'D': -3.500,'E': -3.500,'F':  2.800,'G': -0.400,'H': -3.200,'I':  4.500,'K': -3.900,'L':  3.800,'M':  1.900,'N': -3.500,'P': -1.600,'Q': -3.500,'R': -4.500,'S': -0.800,'T': -0.700,'V':  4.200,'W': -0.900,'Y': -1.300 }
+roseman_hydropathicity = {'A':  0.390,'C':  0.250,'D': -3.810,'E': -2.910,'F':  2.270,'G':  0.000,'H': -0.640,'I':  1.820,'K': -2.770,'L':  1.820,'M':  0.960,'N': -1.910,'P':  0.990,'Q': -1.300,'R': -3.950,'S': -1.240,'T': -1.000,'V':  1.300,'W':  2.130,'Y':  1.470 }
+grantham_polarity = {'A':  8.100,'C':  5.500,'D': 13.000,'E': 12.300,'F':  5.200,'G':  9.000,'H': 10.400,'I':  5.200,'K': 11.300,'L':  4.900,'M':  5.700,'N': 11.600,'P':  8.000,'Q': 10.500,'R': 10.500,'S':  9.200,'T':  8.600,'V':  5.900,'W':  5.400,'Y':  6.200 }
+zimmerman_polarity = {'A':  0.000,'C':  1.480,'D': 49.700,'E': 49.900,'F':  0.350,'G':  0.000,'H': 51.600,'I':  0.130,'K': 49.500,'L':  0.130,'M':  1.430,'N':  3.380,'P':  1.580,'Q':  3.530,'R': 52.000,'S':  1.670,'T':  1.660,'V':  0.130,'W':  2.100,'Y':  1.610 }          
+zimmerman_bulkiness = {'A': 11.500,'C': 13.460,'D': 11.680,'E': 13.570,'F': 19.800,'G':  3.400,'H': 13.690,'I': 21.400,'K': 15.710,'L': 21.400,'M': 16.250,'N': 12.820,'P': 17.430,'Q': 14.450,'R': 14.280,'V': 21.570,'S':  9.470,'T': 15.770,'W': 21.670,'Y': 18.030 }
+
+# Creates feature vector containing properties used in SVM model
+# inputs:
+#        seq:   string containing amino acid sequence  
+# output:
+#        numpy.array feature vector containing 35 properties of amino acid sequence 
+def calc_props(seq = ''):
+  """
+  FEATURE VECTOR
+  1   Length
+  2   Molecular weight
+  3   Number of non-polar hydrophobic residues
+  4   Number of polar hydrophilic residues
+  5   Number of uncharged polar hydrophilic residues
+  6   Number of charged polar hydrophilic residues
+  7   Number of positively charged polar hydrophilic residues
+  8   Number of negatively charged polar hydrophilic residues
+  9   Hydrophobicity-Eisenberg scale (Eisenberg et al., 1984)
+  10  Hydrophilicity-Hopp-Woods scale (Hopp and Woods, 1981)
+  11  Hydrophobicity-Kyte-Doolittle (Kyte and Doolittle, 1982)
+  12  Hydropathicity-Roseman scale (Roseman, 1988)
+  13  Polarity-Grantham scale (Grantham, 1974)
+  14  Polarity-Zimmerman scale (Zimmerman et al., 1968)
+  15  Bulkiness (Zimmerman et al., 1968)
+  16-35   Amino acid singlet counts in order: ACDEFGHIKLMNPQRSTVWY
+  """
+
+  if seq:
+    props = [ float(len(seq)),
+            mass_from_sequence(seq),
+            sum([1.0 for aa in seq if aa in non_polar_hydrophobic]),
```

Comment:
should call set() around these sets of peptides. This will turn these operations from O(n^2) to O(n), because set memebership checking is constant-time, but list membership checking is linear